File: 0302-smallest-rectangle-enclosing-black-pixels/0302-smallest-rectangle-enclosing-black-pixels.py
class Solution:
    def minArea(self, image: List[List[str]], x: int, y: int) -> int:
        m, n = len(image), len(image[0])
        top, bot = x, x
        left, right = y, y
        for i in range(m):
            for j in range(n):
                if image[i][j] == '1':
                    top = min(top, i)
                    bot = max(bot, i + 1)
                    left = min(left, j)
                    right = max(right, j + 1)
        
        return (bot - top) * (right - left)
        

File: 0302-smallest-rectangle-enclosing-black-pixels/0302-smallest-rectangle-enclosing-black-pixels.cpp
class Solution {
public:
    int minArea(vector<vector<char>>& image, int x, int y) {
        int ans = 0;
        int left = y, right = y;
        int top = x, bot = x;
        for (int r = 0; r < image.size(); ++r){
            for (int c = 0; c < image[0].size(); ++c){
                if (image[r][c] == '1'){
                    top = min(top, r);
                    bot = max(bot, r + 1);
                    left = min(left, c);
                    right = max(right, c + 1);
                }
            }
        }

        return (right - left) * (bot - top);
    }
};